ZAPD/ZRoom/Commands/SetMinimapChests.cpp

```diff
@@ -0,0 +1,101 @@
+#include "SetMinimapChests.h"
+#include "../../BitConverter.h"
+#include "../../Globals.h"
+#include "../../StringHelper.h"
+#include "../../ZFile.h"
+#include "../ZRoom.h"
+
+using namespace std;
+
+SetMinimapChests::SetMinimapChests(ZRoom* nZRoom, std::vector<uint8_t> rawData, int rawDataIndex)
+	: ZRoomCommand(nZRoom, rawData, rawDataIndex)
+{
+	int numChests = rawData[rawDataIndex + 1];
+	segmentOffset = BitConverter::ToInt32BE(rawData, rawDataIndex + 4) & 0x00FFFFFF;
```

Comment:
Try to use the `GETSEGOFFSET` macro or the `Seg2Filespace` function in situations like this.
```suggestion
	segmentOffset = GETSEGOFFSET(BitConverter::ToInt32BE(rawData, rawDataIndex + 4));
```